### PR TITLE
Support Nacos auth

### LINKED ISSUE #5665: SkyWalking Nacos implementation of Coordinator/Dynamic Configuration authentication
Support authentication in Nacos implementation of coordinator and dynamic configuration,

https://nacos.io/en-us/blog/nacos%201.2.0%20guide.html

### PR DESCRIPTION
**Pull request by xbkaishui**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
resolved #5665 
___
### Bug fix
- Bug description.
Nacos auth method
- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Review by wu-sheng — COMMENTED** [review]
application.yml and configuration-vocabulary.md should be updated as well. Also, consider updating the UT for the Nacos.

**Comment by wu-sheng** [comment]
Also, we have a Nacos related plugin in the dynamic configuration plugin, please consider to update that too.

**Comment by xbkaishui** [comment]
Changed, @wu-sheng  I don't know how to write test with password scenario, can you give some suggestions ? thanks

**Comment by wu-sheng** [comment]
Doesn't the Nacos image support password and username setup through the system env? There are already integration tests written, you just need to update thems.

**Comment by xbkaishui** [comment]
I am afraid not. [nacos-server](https://hub.docker.com/r/nacos/nacos-server ) list all the envs.  I will double check later

**Comment by kezhenxu94** [comment]
> I am afraid not. [nacos-server](https://hub.docker.com/r/nacos/nacos-server) list all the envs. I will double check later

Not supported by Nacos, https://github.com/nacos-group/nacos-docker/issues/92 

@ITGcx Can you try this branch and test whether it works at your side? Thanks

**Comment by xbkaishui** [comment]
Hi @kezhenxu94  I think I found a way to set the auth. will test and let you know

**Comment by xbkaishui** [comment]
The default nacos user is nacos, I test in my local docker env. it is ok now

` docker run --name nacos-standalone -e MODE=standalone  -p 8848:8848 -d nacos/nacos-server:1.3.1
`

File: oap-server/server-configuration/configuration-nacos/src/test/resources/application.yml [hunk 1]
```diff
@@ -33,3 +33,9 @@ configuration:
     period: 1
     # the name of current cluster, set the name if you want to upstream system known.
     clusterName: "default"
+      # Nacos auth username
+    username: 'nacos'
```
Comment (wu-sheng) [inline oap-server/server-configuration/configuration-nacos/src/test/resources/application.yml:37]:
```suggestion
    # Nacos auth username
    username: 'nacos'
```

File: oap-server/server-cluster-plugin/cluster-nacos-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/nacos/ClusterModuleNacosProvider.java [hunk 2]
```diff
@@ -63,6 +65,17 @@ public void prepare() throws ServiceNotProvidedException, ModuleStartException {
...
             properties.put(PropertyKeyConst.SERVER_ADDR, config.getHostPort());
             properties.put(PropertyKeyConst.NAMESPACE, config.getNamespace());
+            if (StringUtil.isNotEmpty(config.getUsername()) && StringUtil.isNotEmpty(config.getAccessKey())) {
+                throw new ModuleStartException("Nacos Auth method should choose either username or accessKey, not both");
+            }
+            if (StringUtil.isNotEmpty(config.getUsername())) {
+                properties.put(PropertyKeyConst.USERNAME, config.getUsername());
+                properties.put(PropertyKeyConst.PASSWORD, config.getPassword());
+            }
+            if (StringUtil.isNotEmpty(config.getAccessKey())) {
```
Comment (wu-sheng) [inline oap-server/server-cluster-plugin/cluster-nacos-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/nacos/ClusterModuleNacosProvider.java:75]: Should be `else if`?

File: oap-server/server-configuration/configuration-nacos/src/main/java/org/apache/skywalking/oap/server/configuration/nacos/NacosConfigWatcherRegister.java [hunk 3]
```diff
@@ -56,6 +58,14 @@ public NacosConfigWatcherRegister(NacosServerSettings settings) throws NacosExce
         final Properties properties = new Properties();
         properties.put(PropertyKeyConst.SERVER_ADDR, serverAddr + ":" + port);
         properties.put(PropertyKeyConst.NAMESPACE, settings.getNamespace());
+        if (StringUtil.isNotEmpty(settings.getUsername())) {
+            properties.put(PropertyKeyConst.USERNAME, settings.getUsername());
+            properties.put(PropertyKeyConst.PASSWORD, settings.getPassword());
+        }
+        if (StringUtil.isNotEmpty(settings.getAccessKey())) {
```
Comment (wu-sheng) [inline oap-server/server-configuration/configuration-nacos/src/main/java/org/apache/skywalking/oap/server/configuration/nacos/NacosConfigWatcherRegister.java:65]: Should be `else if`?

File: oap-server/server-configuration/configuration-nacos/src/test/resources/application.yml (hunk 1 above)
Comment (xbkaishui) [reply oap-server/server-configuration/configuration-nacos/src/test/resources/application.yml]: done

File: oap-server/server-cluster-plugin/cluster-nacos-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/nacos/ClusterModuleNacosProvider.java (hunk 2 above)
Comment (xbkaishui) [reply oap-server/server-cluster-plugin/cluster-nacos-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/nacos/ClusterModuleNacosProvider.java]: done

File: oap-server/server-configuration/configuration-nacos/src/main/java/org/apache/skywalking/oap/server/configuration/nacos/NacosConfigWatcherRegister.java (hunk 3 above)
Comment (xbkaishui) [reply oap-server/server-configuration/configuration-nacos/src/main/java/org/apache/skywalking/oap/server/configuration/nacos/NacosConfigWatcherRegister.java]: done

**Review by wu-sheng — APPROVED** [review]

**Comment by kezhenxu94** [comment]
> The default nacos user is nacos, I test in my local docker env. it is ok now
> 
> `docker run --name nacos-standalone -e MODE=standalone -p 8848:8848 -d nacos/nacos-server:1.3.1`

Authentication is disabled by default but you didn’t enable it(by environment variables), so the test case verifies nothing actually, right?

**Comment by xbkaishui** [comment]
> > The default nacos user is nacos, I test in my local docker env. it is ok now
> > `docker run --name nacos-standalone -e MODE=standalone -p 8848:8848 -d nacos/nacos-server:1.3.1`
> 
> Authentication is disabled by default but you didn’t enable it(by environment variables), so the test case verifies nothing actually, right?

Yes, I also test using auth enabled (see below command) , it is ok too, the default user has no problem with test case. so I didn't  enable it
 `docker run --name nacos-standalone2 -e MODE=standalone -e NACOS_AUTH_ENABLE=true -p 8848:8848 -d nacos/nacos-server:1.3.1`

**Pull request merged**